spec/controllers/admin/processors_controller_spec.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+describe Admin::ProcessorsController, type: :controller do
+  describe '#create' do
+    subject { post :create }
+
+    let(:hodinkee_file) { file_fixture('shophodinkee_first_page.html').read }
+    let(:crown_file) { file_fixture('crownandcaliber_first_page.html').read }
+    let(:bobs_file) { file_fixture('bobswatches_first_page.html').read }
+
+    before do
+      allow_any_instance_of(Browser).to receive(:visit)
+
+      allow_any_instance_of(Browser)
+        .to receive(:visit)
+        .with(url: 'https://shop.hodinkee.com/collections/watches?page=18',
+              tag: '.product-title')
+        .and_return(hodinkee_file)
+
+      allow_any_instance_of(Browser)
+        .to receive(:visit)
+        .with(url: 'https://www.crownandcaliber.com/collections/shop-for-watches?page=54',
+              tag: '.card-title.ng-binding')
+        .and_return(crown_file)
+
+      allow_any_instance_of(Browser)
+        .to receive(:visit)
+        .with(url: 'https://www.bobswatches.com/shop?page=21',
+              tag: '.product-title')
+        .and_return(bobs_file)
+    end
+
+    describe 'with correct pages' do
+      let(:hodinkee_item_file) { file_fixture('shophodinkee_item_page.html').read }
+      let(:hodinkee_created_item) { HodinkeeItem.find_by(model: 'Prospex SLA043 Limited Edition') }
+
+      let(:crown_item_file) { file_fixture('crownandcaliber_item_page.html').read }
+      let(:crown_created_item) { CrownandcaliberItem.find_by(model: 'Datejust') }
+
+      let(:bobs_item_file) { file_fixture('bobswatches_item_page.html').read }
+      let(:bobs_created_item) { BobswatchesItem.find_by(model: 'Sky-Dweller 326934') }
+
+      before do
+        allow_any_instance_of(Browser)
+          .to receive(:visit)
+          .with(url: 'https://shop.hodinkee.com/collections/watches/products/aquaracer-200-blue-dial-on-bracelet',
+                tag: '.vendor')
+          .and_return(hodinkee_item_file)
+
+        allow_any_instance_of(Browser)
+          .to receive(:visit)
+          .with(url: 'https://www.crownandcaliber.com/products/rolex-datejust-126334-10-10-rol-altzu5',
+                tag: '.vendor')
+          .and_return(crown_item_file)
+
+        allow_any_instance_of(Browser)
+          .to receive(:visit)
+          .with(url: 'https://www.bobswatches.com/mens-rolex-gmt-master-ii-116710blnr.html',
+                tag: '.price')
+          .and_return(bobs_item_file)
+      end
+
+      it 'creates items with needed params' do
+        subject
```

Comment:
why do you use `subject`  here? use it in  `before {}`

Reply:
I can't use `subject` in `before {}` here, because in this scope I have test which pass `subject` into block:
     ` it { expect { subject }.to change(Item, :count).by(3) }`

Reply:
1. then you can use `is_expected.to` instead of `expect { subject }.to`
2. use `context do ... end` to isolate different tests